File: introduction/ITP1_11_A-Dice I.java
import java.util.*;
public class Main {
  public static void main(String[] args) {
    Scanner sc = new Scanner(System.in);
    Dice dice = new Dice(sc.nextLine().split(" "));
    char[] chars = sc.nextLine().toCharArray();
    for(char c:chars) dice.roll(c);
  }
}

class Dice {
  private String[] labels = new String[6];
  Dice(String[] labels) {
    this.labels = labels;
  }
  public void roll(char dir) {
    System.out.println(dir);
  }
}

File: introduction/ITP1_11_A-Dice I.rb
class Dice

  def initialize(labels)
    @labels = labels
  end

  def roll(dir)
    a, b, c, d, e, f = @labels
    case dir
    when 'N' then @labels = [b, f, c, d, a, e]
    when 'S' then @labels = [e, a, c, d, f, b]
    when 'E' then @labels = [d, b, a, f, e, c]
    when 'W' then @labels = [c, b, f, a, e, d]
    end
  end

  def top
    @labels[0]
  end

end

dice = Dice.new(gets.split.map(&:to_i))
gets.split('').each do | v |
  dice.roll(v)
end
puts dice.top
